opencti-platform/opencti-graphql/src/manager/ingestionManager.ts

```diff
@@ -204,59 +204,106 @@ const rssExecutor = async (context: AuthContext, turndownService: TurndownServic
-  return { data, addedLast: resultHeaders['x-taxii-date-added-last'] };
+  const params = prepareTaxiiGetParam(ingestion);
+  const { data, headers } = await httpClient.get(url, { params });
+  logApp.info('[OPENCTI-MODULE] Taxii HTTP Get done.', { ingestionId: ingestion.id, moreResponse: data.more, nextResponse: data.next, url, requestParams: params, addedLastHeaderResponse: headers['x-taxii-date-added-last'] });
+  return { data, addedLastHeader: headers['x-taxii-date-added-last'] };
 };
+
 type TaxiiHandlerFn = (context: AuthContext, ingestion: BasicStoreEntityIngestionTaxii) => Promise<void>;
-const taxiiV21DataHandler: TaxiiHandlerFn = async (context: AuthContext, ingestion: BasicStoreEntityIngestionTaxii) => {
-  const { data, addedLast } = await taxiiHttpGet(ingestion);
+
+export const processTaxiiResponse = async (context: AuthContext, ingestion: BasicStoreEntityIngestionTaxii, taxiResponse:TaxiiResponseData) => {
+  const { data, addedLastHeader } = taxiResponse;
   if (data.objects && data.objects.length > 0) {
-    logApp.info(`[OPENCTI-MODULE] Taxii ingestion execution for ${data.objects.length} items`);
-    const bundle = { type: 'bundle', id: `bundle--${uuidv4()}`, objects: data.objects };
+    logApp.info(`[OPENCTI-MODULE] Taxii ingestion execution for ${data.objects.length} items, sending stix bundle to workers.`, { ingestionId: ingestion.id });
+
+    const bundle = { type: 'bundle', id: `bundle--${uuidv4()}`, objects: taxiResponse.data.objects };
     // Push the bundle to absorption queue
     const stixBundle = JSON.stringify(bundle);
     const content = Buffer.from(stixBundle, 'utf-8').toString('base64');
-    await pushToWorkerForSync({ type: 'bundle', applicant_id: ingestion.user_id ?? OPENCTI_SYSTEM_UUID, content, update: true });
+    await pushToWorkerForSync({
+      type: 'bundle',
+      applicant_id: ingestion.user_id ?? OPENCTI_SYSTEM_UUID,
+      content,
+      update: true
+    });
+    const more = data.more || false;
     // Update the state
+    if (more) {
+      // Do not touch to added_after_start
+      await patchTaxiiIngestion(context, SYSTEM_USER, ingestion.internal_id, {
+        current_state_cursor: data.next ? data.next : undefined,
+        taxii_more: more,
+        last_execution_date: now()
+      });
+    } else {
+      // Reset the pagination cursor
+      await patchTaxiiIngestion(context, SYSTEM_USER, ingestion.internal_id, {
+        current_state_cursor: undefined,
+        taxii_more: more,
+        added_after_start: addedLastHeader,
+        last_execution_date: now()
+      });
+    }
+  } else {
+    // Update the last run
     await patchTaxiiIngestion(context, SYSTEM_USER, ingestion.internal_id, {
-      current_state_cursor: data.next ? String(data.next) : undefined,
-      added_after_start: data.next ? ingestion.added_after_start : utcDate(addedLast)
+      last_execution_date: now(),
+      current_state_cursor: undefined, // no data = pagination is ended, reset the cursor.
```

Comment:
why just reset the cursor ? next time we run this taxii feed we will restart from the last added_after_start. I think we should update added_after_start then if we reset the cursor, otherwise we might restart the ingestion and get data we already ingested.